infra.bs

```diff
@@ -282,8 +283,117 @@ a simple wrapper around another one.)
+  "<code>form submission</code>" and true.
+ </ul>
+
+ <p>But, there would be no way to supply a non-default value for the third
+ (<var ignore>exceptionsEnabled</var>) argument, while leaving the second
+ (<var ignore>navigationType</var>) argument as not-given. Additionally, the last of these calls is
+ fairly unclear for readers, as the fact that "true" means "exceptions enabled" requires going back
+ to the algorithm's declaration and counting parameters. Read on for how to fix these issues!
+</div>
+
+<p>Optional named parameters, instead of positional ones, can be used to increase clarity and
+flexibility at the call site. Such parameters are marked up as both variables and definitions, and
+linked to from their call sites.
+
+<div class=example id=example-algorithm-optional-named-params>
+ <p class=allow-2119>To <dfn ignore id=example-navigate-algo-named>navigate</dfn> given a
+ resource <var ignore>resource</var>, an optional string
+ <dfn ignore id=example-navigate-algo-navigationType><var ignore>navigationType</var></dfn>, and an
+ optional boolean
+ <dfn ignore id=example-navigate-algo-exceptionsEnabled><var ignore>exceptionsEnabled</var></dfn>
+ (default false):
+
+ <ol class=brief>
+  <li>&hellip;
+  <li>If <var ignore>navigationType</var> was given, then do something with
+  <var ignore>navigationType</var>.
+  <li>&hellip;
+ </ol>
+
+ <p>Call sites would then look like one of:
+
+ <ul class=brief>
+  <li><a href=#example-navigate-algo-named>Navigate</a> to <var ignore>resource</var>.
+
+  <li><a href=#example-navigate-algo-named>Navigate</a> to <var ignore>resource</var> with
+  <a href=#example-navigate-algo-navigationType><var ignore>navigationType</var></a> set to
+  "<code>form-submission</code>".
+
+  <li><a href=#example-navigate-algo-named>Navigate</a> to <var ignore>resource</var> with
+  <a href=#example-navigate-algo-exceptionsEnabled><var ignore>exceptionsEnabled</var></a> set to
+  true.
+
+  <li><a href=#example-navigate-algo-named>Navigate</a> to <var ignore>resource</var> with
+  <a href=#example-navigate-algo-navigationType><var ignore>navigationType</var></a> set to
+  "<code>form-submission</code>" and
+  <a href=#example-navigate-algo-exceptionsEnabled><var ignore>exceptionsEnabled</var></a> set to
+  true.
+ </ul>
+</div>
+
+<p class=note>Note how within the algorithm steps, the argument value is not linked to the parameter
+declaration; it remains just a variable reference. Linking to the parameter declaration is done only
+at the call sites.
+
+<p>Non-optional named parameters may also be used, using the same convention of marking them up as
+both variables and definitions, and linking to them from call sites. However, one should default to
+making non-optional parameters positional, for simplicity and uniformity. Non-optional named
+parameters should only be used when they significantly improve clarity, and even then it might be
```

Comment:
FWIW, I think `Call foo with 3` and `Call foo with "navigate"` are not much better than `Call foo with false`, and Infra should encourage people to name those non-optional parameters. Perhaps:

```suggestion
<p>Non-optional parameters may also be named, using the same convention of marking them up as
both variables and definitions, and linking to them from call sites. Do this when it improves
clarity at call sites. It is also
```